src/AST.cc

```diff
@@ -23,6 +24,9 @@ bool Location::isNone() const{
 
 std::string Location::toString() const{
 	if(this->isNone()) return "location unkown";
+	if(mainfile && (*path)!=(*mainfile)){
+		return "in file "+boostfs_uncomplete((*path), (*mainfile).parent_path()).generic_string()+ ", "+"line " + std::to_string(this->firstLine());
```

Comment:
I find all these parentheses hard to read; `(*path)` => `*path`
also, `(*mainfile).` => `mainfile->`